api.ts

```diff
@@ -2203,3 +2296,49 @@ export function maybeRewriteConnectionForFormula<
   }
   return formula;
 }
+
+// This helper method finds any inline autocomplete functions in a static sync table schema.
+// These functions will need to be extracted into the "namedAutocompletes" property on the sync
+// table and replaced with strings.
+function listPropertiesWithAutocompleteFunctions(schema: ObjectSchemaDefinition<string, string>): string[] {
+  const result: string[] = [];
+  for (const propertyName of Object.keys(schema.properties)) {
+    const {autocomplete} = schema.properties[propertyName];
+    if (!autocomplete) {
+      continue;
+    }
+    if (typeof autocomplete !== 'function') {
+      continue;
+    }
+
+    result.push(propertyName);
+  }
+  return result;
+}
+
+// Finds any inline autocomplete functions within the inputSchema and replaces them
+// with strings references into the returned namedAutocompletes.
+function replaceInlineAutocompleteFunctionsWithNamedAutocompleteFunctions({
+  inputSchema, // DO NOT MUTATE inputSchema!
```

Comment:
Is there anything we do with `Readonly<T>` or something to help enforce this?

Reply:
I did a quick test and it seems like `Readonly` only protects the top-level properties and not deeply nested ones?

I've added in a `Readonly<>` wrapper just to emphasize to the reader that they shouldn't change it, but I haven't removed the comment

Reply:
we can also freeze the object of pass in a copy. not sure if it's worth.

Reply:
I think deep freezing an input parameter in-place would technically be a caller-visible mutation to that parameter, since afterwards the caller wouldn't be able to mutate the value either.